fdbserver/TLogServer.actor.cpp

```diff
@@ -2025,6 +2025,9 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 	reply.messages = messagesValue;
 	reply.end = endVersion;
 	reply.onlySpilled = onlySpilled;
+	if (reqTag.locality == tagLocalityTxs) {
+		reply.popped = poppedVer;
+        }
```

Comment:
The contract that I recall is that popped is only set in the reply when there is no data, and the peek reply is informing that there's no data because the popped version is higher than the peek version.  It does feel a little odd to need to break that contract only for transaction state store.

If you really do need to do this change, please do verify that OldTLog_*.cpp files don't need the same change, and that excluding tag==txsTag from this is correct as well.

Reply:
This is my interpretation of what is happening.

* test has 4 dcs (replica 3 on locals), dc3 is satellite for dc0.
* cycle test runs for some time up to version X. 
* master is moved from dc0 to dc1
* Then dc0, dc2 crash and dc3 rebooted
* dc1 does not have versions 0-X, so we must go to dc3
* dc3 starts machines but tlogs are never assigned to them, as we are blocked on rebuilding the txn state store.

I suspected there may be a contract, so limit my change to the txn tag.

Reply:
I think that when we calculate the best log group to pull old logs from, we do not alternate between log groups. So if one of the log groups is on a crashed machine it will be selected repeatedly and we block.